File: src/main/java/com/mbayou/kick4k/api/ApiResponse.java
package com.mbayou.kick4k.api;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class ApiResponse<T> {
    private final T data;
    private final String message;

    @JsonCreator
    public ApiResponse(@JsonProperty("data") T data, @JsonProperty("message") String message) {
        this.data = data;
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public String getMessage() {
        return message;
    }

    public boolean isSuccess() {
        return data != null;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/api/ApiResponse.kt
package com.mbayou.kick4k.api

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class ApiResponse<T> @JsonCreator constructor(
    @JsonProperty("data") val data: T?,
    @JsonProperty("message") val message: String?,
) {
    fun isSuccess(): Boolean = data != null
}
